File: src/test/resources/FirstNonRepeatedChar.java
import java.util.*;
import com.jalgoarena.type.*;

public class Solution {
    /**
     * @param str Input string
     * @return  First non-duplicate character or null if absent
     */
    public String findFirstNonRepeatedChar(String str) {
        if (str == null || str.length() == 0) {
            return null;
        }

        char[] chars = str.toCharArray();
        Map<Character, Integer> register = new HashMap<>();

        for(char c : chars) {
            if (register.containsKey(c)) {
                register.put(c, 2);
            } else {
                register.put(c, 1);
            }
        }

        for (char c : chars) {
            if (register.get(c) == 1) return Character.toString(c);
        }

        return null;
    }
}

File: src/test/resources/FirstNonRepeatedChar.kt
import java.util.*

class FirstNonRepeatedChar {
    /**
     * @param str Input string
     * *
     * @return  First non-duplicate character or null if absent
     */
    fun findFirstNonRepeatedChar(str: String?): String? {
        if (str == null || str.isEmpty()) {
            return null
        }

        val chars = str.toCharArray()
        val register = HashMap<Char, Int>()

        for (c in chars) {
            if (register.containsKey(c)) {
                register.put(c, 2)
            } else {
                register.put(c, 1)
            }
        }

        return chars
                .firstOrNull { register[it] === 1 }
                ?.let { Character.toString(it) }
    }
}
